Remotion/Data/DomainObjects.Validation/DomainObjectAttributesBasedValidationPropertyRuleReflector.cs

```diff
@@ -170,15 +170,22 @@ public IEnumerable<IPropertyValidator> GetRemovablePropertyValidators ()
             _validationMessageFactory);
       }
 
-      if (!_domainModelConstraintProvider.IsNullable(_implementationPropertyInformation)
-          && typeof(IEnumerable).IsAssignableFrom(_implementationProperty.PropertyType)
-          && !ReflectionUtility.IsObjectList(_implementationProperty.PropertyType)
-          && !ReflectionUtility.IsIObjectList(_implementationProperty.PropertyType))
+      if (!_domainModelConstraintProvider.IsNullable(_implementationPropertyInformation))
       {
-        yield return PropertyValidatorFactory.Create(
-            _implementationPropertyInformation,
-            parameters => new NotEmptyValidator(parameters.ValidationMessage),
-            _validationMessageFactory);
+        if (typeof(string).IsAssignableFrom(_implementationProperty.PropertyType))
```

Comment:
In this case, I think the logic is more readable with the deeper nesting, because the separate IsNullable-check expresses that the entire nested block is about properties that are not nullable.

Reply:
@a-ctor you agree, too?

Reply:
I think the nested version is more readable.